include/crow/http_connection.h

```diff
@@ -404,6 +406,8 @@ namespace crow
             {
                 do_write_general();
             }
+
+            queue_length--;
```

Comment:
It would probably be best to make this a part of `check_destroy()`, since that's the place where the connection is freed from memory.

Reply:
I've also noticed a memory leak in the connection not being deleted when not using `do_write()`, this would probably need to be addressed before this PR is merged

Reply:
Thanks! I didn't know where the connection was ending. It's done now.

Let me know when the leak is fixed or if I can help.